analyzer.go

```diff
@@ -333,12 +333,25 @@ func (gosec *Analyzer) load(pkgPath string, conf *packages.Config) ([]*packages.
 		return []*packages.Package{}, fmt.Errorf("importing dir %q: %w", pkgPath, err)
 	}
 
-	var packageFiles []string
-	for _, filename := range basePackage.GoFiles {
-		packageFiles = append(packageFiles, path.Join(pkgPath, filename))
+	var goModFile string
+	goModDir := abspath[0 : len(abspath)-len(pkgPath)] // get root dir
+	if modPkgs, err := packages.Load(&packages.Config{Mode: packages.NeedModule, Dir: abspath}, abspath); err == nil && len(modPkgs) == 1 {
```

Comment:
Is this only loading the module files, or does parse all the package as well? Please could you double check? I want to avoid to introduce performance issues?

Reply:
Why is this `1`? What if there are more than 1 module files?

Reply:
I'll double check on performance but my assumption was with only NeedModule is that it skips the remaining parsing , the check is for `1` given that it should only find a single go.mod for the given source file - or there are go files with different package identifiers in the same directory (which is not valid).

Reply:
Added benchmarks, see below